server/src/main/java/org/elasticsearch/common/geo/GeoPoint.java

```diff
@@ -126,7 +125,12 @@ public GeoPoint resetFromIndexableField(IndexableField field) {
     }
 
     public GeoPoint resetFromGeoHash(String geohash) {
-        final long hash = mortonEncode(geohash);
+        final long hash;
+        try {
+            hash = mortonEncode(geohash);
+        } catch (IllegalArgumentException ex) {
+            throw new ElasticsearchParseException(ex.getMessage(), ex);
+        }
```

Comment:
why don't we just use the IAE?

Reply:
All other formatting issues are throwing `ElasticsearchParseException` at the moment, so I have decided to stick with it for consistency sake. It also makes it easier to ignore in `parseGeoPointIgnoringMalformed` and `parseGeoPointStringIgnoringMalformed` later on if mapping has `ignore_malformed` set to `true`.